test_pull_requests

```diff
@@ -359,6 +361,78 @@ def compose_bot_comment(prefix, comment_data={})
   updated_comment
 end
 
+class FlushableStore
+  def initialize(c_file=nil)
+    @cache_file = nil
+    @cache = {}
+    if c_file
+      @cache_file = File.expand_path(c_file)
+      if File.exists? @cache_file
+        if File.writable? @cache_file
+          begin
+            @cache = Marshal.load IO.read(cache_file)
+          rescue
+            $stderr.puts "  *** Starting with empty cache file at #{c_file}"
+          end
+        else
+          $stderr.puts "  *** Specified cache file #{@cache_file} is not writable. Cache won't persist between runs."
+          @cache_file = nil
+        end
+      else
+        begin
+          FileUtils.touch @cache_file
+        rescue
+          $stderr.puts "  *** Can't create fresh cache file at #{@cache_file}. Cache won't persist between runs."
+          @cache_file = nil
+        end
+      end
+    end
+  end
+
+  def read(key)
+    @cache[key]
+  end
+
+  def delete(key)
+    @cache.delete(key)
+  end
+
+  def write(key, value)
+    @cache[key] = value
+  end
+
+  def flush
+    if @cache_file
+      IO.write(@cache_file, Marshal.dump(@cache))
+    else
+      $stderr.puts "  WARNING: Cache can't be flushed to disk, and won't persist to future runs."
+    end
+  end
+end
+
+class CacheStatsInstrumenter
+
+  attr_reader :stats
+
+  def initialize()
+    @stats = {:hits => 0, :misses => 0, :bypasses => 0}
+  end
+
+  def instrument(name, payload)
+    if name == 'process_request.http_cache.faraday'
+      cache_status = payload[:cache_status]
+      case cache_status
+      when :fresh, :valid
```

Comment:
I guess the argument just is -- we have the data, let's just show it. I don't see a huge reason to reduce the granularity of the data we have. Less work just to print it.